api/src/test/java/org/openmrs/api/VisitServiceTest.java

```diff
@@ -418,6 +418,37 @@ public void purgeVisit_shouldEraseTheVisitFromTheDatabase() {
 		assertEquals(originalSize - 1, vs.getVisits(null, null, null, null, null, null, null, null, null, true, true).size());
 	}
 	
+	/**
+	 * @see VisitService#purgeVisit(Visit)
+	 */
+	@Test
+	public void purgeVisit_shouldFailIfVisitIsUnsaved() {
```

Comment:
please rename the method to `purgeVisit_shouldReturnWithoutVoidingIfVisitIsUnsaved` since it doesnt fail but just wont do anything.

and change your assert to

```java
assertEquals(originalSize, xxx)
```

this will give us better error messages when the assertion fails

and I also like to use the AAA style, arrange act assert

so first setup (arrange)

than act (in this case purgeVisit)

and then the assertions

all separated by a newline